tsdb/index/postings.go

```diff
@@ -402,6 +402,23 @@ func (p *MemPostings) addFor(id storage.SeriesRef, l labels.Label) {
 }
 
 func (p *MemPostings) PostingsForLabelMatching(ctx context.Context, name string, match func(string) bool) Postings {
+	if match == nil {
```

Comment:
I was going to ask for this to be documented on the interface, but since it's a breaking change, I would like to ask you to define a completely new method for this.

If we keep the method signature the same, a downstream implementation may not be aware of the change, and it would try to call `match()` and would panic.

If we add a new method, then if there's a downstream implementation, they will need to explicitly implement the new method. If there isn't, they won't be affected.

So I would suggest something like `PostingsForAllLabelValues(ctx context.Contex, name string) Postings`

Reply:
Implemented your suggestion @colega, i.e. `PostingsForAllLabelValues`.